src/database/DBOperationQueue.cpp

```diff
@@ -0,0 +1,61 @@
+#include "DBOperationQueue.h"
+
+#include <algorithm>
+
+using namespace std;
+
+DBOperation *DBOperationQueue::get_next_operation()
+{
+    if(!m_queue.size()) {
+        return nullptr;
+    }
+
+    DBOperation *op = m_queue.front();
+
+    if(can_operation_start(op)) {
+        m_running_operations.insert(op);
+        m_queue.pop();
+        return op;
```

Comment:
The function name is a bit misleading. The name implies we're getting the next operation in the list, yet this section also starts it (something that should be in begin_operation?) if we have a legible operation.